modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2844,8 +2844,15 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
             codegenParameter.vendorExtensions.putAll(parameter.getExtensions());
         }
 
-        if (parameter.getSchema() != null) {
-            Schema parameterSchema = ModelUtils.unaliasSchema(this.openAPI, parameter.getSchema());
+        if (parameter.getSchema() != null || parameter.getContent() != null) {
+            Schema parameterSchema = null;
+            if (parameter.getSchema() != null) {
+                parameterSchema = ModelUtils.unaliasSchema(this.openAPI, parameter.getSchema());
+            } else if (parameter.getContent().values().stream().findFirst().isPresent()) {
+                MediaType mediaType = parameter.getContent().values().stream().findFirst().get();
+                parameterSchema = ModelUtils.getReferencedSchema(openAPI, mediaType.getSchema());
+            }
```

Comment:
@turtle11311 What about adding an else clause and log an error?

Also can you please share the spec to reproduce the issue?